### PR TITLE
sparse-merkle-tree bench update

### PR DESCRIPTION
**Pull request by zjma**
### Description

Use jemalloc for sparse merkle tree benchmark, and a few other tweaks.

### CODE REVIEW

File: storage/scratchpad/benches/sparse_merkle.rs [hunk 1]
```diff
@@ -13,22 +13,19 @@ use itertools::zip_eq;
...
 
 impl Block {
-    fn updates(&self) -> Vec<Vec<(HashValue, Option<&StateValue>)>> {
-        self.updates
-            .iter()
-            .map(|small_batch| small_batch.iter().map(|(k, v)| (*k, v.as_ref())).collect())
-            .collect()
-    }
-
     fn updates_flat_batch(&self) -> Vec<(HashValue, Option<&StateValue>)> {
```
Comment (msmouse) [inline storage/scratchpad/benches/sparse_merkle.rs:27]: nit: rename

File: storage/scratchpad/benches/sparse_merkle.rs [hunk 2]
```diff
@@ -45,7 +42,7 @@ impl Group {
             let block_size = block.updates.len();
             let one_large_batch = block.updates_flat_batch();
 
-            group.throughput(Throughput::Elements(block_size as u64));
+            group.throughput(Throughput::Elements((block_size * 2) as u64));
```
Comment (msmouse) [inline storage/scratchpad/benches/sparse_merkle.rs:45]: don't need to x2?

File: storage/scratchpad/benches/sparse_merkle.rs [hunk 3]
```diff
@@ -167,9 +164,9 @@ impl Benches {
         rng: &mut StdRng,
         keys: &[HashValue],
         block_size: usize,
-    ) -> Vec<Vec<(HashValue, Option<StateValue>)>> {
-        std::iter::repeat_with(|| vec![Self::gen_update(rng, keys), Self::gen_update(rng, keys)])
-            .take(block_size)
+    ) -> Vec<(HashValue, Option<StateValue>)> {
+        std::iter::repeat_with(|| Self::gen_update(rng, keys))
+            .take(block_size * 2)
```
Comment (msmouse) [inline storage/scratchpad/benches/sparse_merkle.rs:169]: I think it's more straightforward not to x2

File: storage/scratchpad/benches/sparse_merkle.rs (hunk 1 above)
Comment (zjma) [reply storage/scratchpad/benches/sparse_merkle.rs]: Fixed.

File: storage/scratchpad/benches/sparse_merkle.rs (hunk 2 above)
Comment (zjma) [reply storage/scratchpad/benches/sparse_merkle.rs]: Fixed.

File: storage/scratchpad/benches/sparse_merkle.rs (hunk 3 above)
Comment (zjma) [reply storage/scratchpad/benches/sparse_merkle.rs]: Fixed.

**Pull request merged**